build.cmd

```diff
@@ -471,41 +471,29 @@ if %__BuildCrossArchNative% EQU 1 (
     if not exist "%__CrossCompIntermediatesDir%" md "%__CrossCompIntermediatesDir%"
     if defined __SkipConfigure goto SkipConfigureCrossBuild
 
-    pushd "%__CrossCompIntermediatesDir%"
     set __CMakeBinDir=%__CrossComponentBinDir%
     set "__CMakeBinDir=!__CMakeBinDir:\=/!"
     set __ExtraCmakeArgs="-DCLR_CROSS_COMPONENTS_BUILD=1" "-DCLR_CMAKE_TARGET_ARCH=%__BuildArch%" "-DCLR_CMAKE_TARGET_OS=%__BuildOS%" "-DCLR_CMAKE_PACKAGES_DIR=%__PackagesDir%" "-DCLR_CMAKE_PGO_INSTRUMENT=%__PgoInstrument%" "-DCLR_CMAKE_OPTDATA_VERSION=%__PgoOptDataVersion%" "-DCLR_CMAKE_PGO_OPTIMIZE=%__PgoOptimize%" "-DCMAKE_SYSTEM_VERSION=10.0"
-    call "%__SourceDir%\pal\tools\gen-buildsys-win.bat" "%__ProjectDir%" %__VSVersion% %__CrossArch% !__ExtraCmakeArgs!
+    call "%__SourceDir%\pal\tools\gen-buildsys.cmd" "%__ProjectDir%" "%__CrossCompIntermediatesDir%" %__VSVersion% %__CrossArch% !__ExtraCmakeArgs!
+    
+    if not !errorlevel! == 0 (
+        echo %__ErrMsgPrefix%%__MsgPrefix%Error: failed to generate native component build project!
+        exit /b 1
+    )
     @if defined _echo @echo on
-    popd
 
 :SkipConfigureCrossBuild
-    if not exist "%__CrossCompIntermediatesDir%\install.vcxproj" (
-        echo %__ErrMsgPrefix%%__MsgPrefix%Error: failed to generate cross-arch components build project!
+    if not exist "%__CrossCompIntermediatesDir%\CMakeCache.txt" (
+        echo %__ErrMsgPrefix%%__MsgPrefix%Error: unable to find generated native component build project!
         exit /b 1
     )
 
     if defined __ConfigureOnly goto SkipCrossCompBuild
 
-    set __BuildLogRootName=Cross
```

Comment:
I wonder how CMake generates the logs. We had specific verbosity settings that we seem to be losing with your change.

Reply:
The logs here were generated by MSBuild under the hood when MSBuild is run.

Reply:
The actual writing out of the logs to a file is something I'll make sure we still get if we feel that it's important (it's pretty easy to just pull down the job logs from AzDO directly, so writing them out separately might not be necessary).

Reply:
I find the log files useful when looking into build issues. The Windows console screen buffer limit is quite small and it is easier to view the log files in a text editor to search for some strings etc.
E.g. the test build log for pri-1 tests has about 768000 lines.

It seems we could pass in the msbuild options at the end of the cmake --build command line (just adding `--` followed by the msbuild options).

Reply:
I'll add them back.